subcommands/common_config.go

```diff
@@ -220,22 +225,62 @@ func SetUpdatesConfig(opts *SetUpdatesConfigOptions, reportedTag string, reporte
 	}
 
 	changed := false
-	if opts.UpdateApps != "" && configuredApps != opts.UpdateApps {
-		if strings.TrimSpace(opts.UpdateApps) == "," {
-			opts.UpdateApps = ""
+	if opts.UpdateAppsGiven || opts.InheritApps {
+
+		if opts.UpdateTagGiven && opts.UpdateTag != "" {
+			configuredTag = opts.UpdateTag
+		}
+		if opts.InheritApps && sota.Has("pacman.compose_apps") {
+			fmt.Printf("Changing apps from: [%s] -> inherit from parent\n", configuredApps)
+			DieNotNil(sota.Delete("pacman.docker_apps"))
+			DieNotNil(sota.Delete("pacman.compose_apps"))
+			changed = true
+		} else if configuredApps != opts.UpdateApps && opts.UpdateAppsGiven {
+			if strings.TrimSpace(opts.UpdateApps) == "," {
+				targets, err := api.TargetsList(factory)
+				DieNotNil(err)
+				version := 0
+				var final_apps []string
+				for _, target := range targets {
+					custom, err := api.TargetCustom(target)
+					DieNotNil(err)
+					set := make(map[string]bool)
+					var apps []string
+					if slices.Contains(custom.Tags, configuredTag) {
+						for app := range custom.ComposeApps {
+							if _, ok := set[app]; !ok {
+								apps = append(apps, app)
+							}
+						}
+						if v, err := strconv.Atoi(custom.Version); err == nil {
+							if v > version {
+								final_apps = apps
+								version = v
+							}
+						}
+					}
+				}
+				opts.UpdateApps = strings.Join(final_apps, ",")
```

Comment:
IIUC this is a huge change in behavior.

1. In the past `--apps=","` was used to tell the device to run no apps, and this PR changes it to the opposite: run app apps.

2. Furthermore, it does that in an IMHO wrong way:
- a true way to tell to run all known apps is to set apps to an empty string, this way the device will always run all applications known at a time of the update.
- with this change, the device will be stick to run the apps known at a time of config change; when the list of apps change with the future updates - the device would not update to a new apps list automatically.

So, I _believe_ that:
- for 1 we must revert the original behavior to be backward compatible, i.e. `--apps=,` should result into "no apps".
- for 2 we should find a way to set the `pacman.docker_apps=""` and verify if it overrides the upper level `sota.toml` values properly. If not, maybe your idea to set the device to all _currently known_ apps is OK, but we must use a different _special value_ than a comma for this; maybe even a separate flag like `--all-apps`.

Reply:
@StealthyCoder for now, please, ignore my comment before @mike-sul gives a definite answer.
I might make some _too early_ conclusions.

Reply:
@vkhoroz 
> But, what will happen if sota.toml has pacman.compose_apps=,? Will it run "no apps" or "all apps"?

As far as I can tell through testing, it does the run no apps. Same for "" or " ", but I have not tried leaving it open. `pacman.compose_apps=` . So I will try that. 

I am not sure how to send that through via Golang but that is something different.

Reply:
As per the offline email thread, let's just remove this logic to select "all apps".
It is a new "potential" feature, which is rather ambiguous, as some users would expect it to include "all future apps".

The previous behavior for "," value was indeed to set "no apps".
So, let's just preserve this behavior _as is_.

An option to introduce an explicit `--no-apps` flag (and prohibit a "," here) looks appealing, but it would require too much effort to update docs/blogs/etc.

Reply:
@doanac I _guess_ you should agree to this _being conservative_ proposal.